projects/vdk-plugins/vdk-data-sources/src/vdk/plugin/data_sources/factory.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Any
+from typing import Collection
+from typing import Dict
+from typing import List
+from typing import Type
+
+from vdk.plugin.data_sources import config
+from vdk.plugin.data_sources.data_source import IDataSource
+from vdk.plugin.data_sources.data_source import (
+    IDataSourceConfiguration,
+)
+
+
+def data_source(name: str, config_class: Type[IDataSourceConfiguration]):
+    def inner_decorator(data_source_class: Type[IDataSource]):
+        data_source_class.__data_source_name__ = name
+        data_source_class.__data_source__config__class__ = config_class
+        return data_source_class
+
+    return inner_decorator
+
+
+@dataclass
+class DataSourceRegistryItem:
```

Comment:
Is the user who is writing an ingestion job suppose to know about this class?

Reply:
Not really